houndify/houndify_client.go

```diff
@@ -162,13 +162,18 @@ func (c *Client) TextSearch(textReq TextRequest) (string, error) {
 		fmt.Println(bodyStr)
 	}
 
-	// update with new conversation state
-	if c.enableConversationState {
-		newConvState, err := parseConversationState(bodyStr)
-		if err != nil {
-			return bodyStr, errors.New("unable to parse new conversation state from response")
+	//don't try to parse out conversation state from a bad response
+	if resp.StatusCode < 400 {
```

Comment:
Just a kind suggestion, could we also reverse conditional it so
```go
if resp.StatusCode >= 400 {
   return bodyStr, errors.New(resp.Status)
}
```
then carry on normally, with a minimal diff.